subprojects/packagefiles/cfitsio/meson.build

```diff
@@ -0,0 +1,270 @@
+  elif cc.get_id() == 'msvc'
+    add_project_arguments('-D__SSE2__=1', '-D__SSSE3__=1', language: 'c')
+  endif
+endif
+
+if cc.has_header_symbol('unistd.h', 'ftruncate')
+  add_project_arguments('-DHAVE_UNISTD_H', '-DHAVE_FTRUNCATE', language: 'c')
+endif
+
+shmem_prefix = [
+  '#include <sys/ipc.h>',
+  '#include <sys/shm.h>',
+  '#include <sys/sem.h>',
+]
+flock_prefix = ['#include <sys/flock.h>']
+
+have_shmem_services = true
+foreach func : ['shmat', 'shmdt', 'shmget', 'semget']
+  have_shmem_services = have_shmem_services and cc.has_function(
+    func,
+    prefix: shmem_prefix,
+  )
+endforeach
+if have_shmem_services
+  add_project_arguments('-DHAVE_SHMEM_SERVICES', language: 'c')
+  if cc.has_type('flock_t', prefix: flock_prefix)
+    add_project_arguments('-DHAVE_FLOCK_T', language: 'c')
+  endif
+  if cc.has_type('union semun', prefix: shmem_prefix)
+    add_project_arguments('-DHAVE_UNION_SEMUN', language: 'c')
+  endif
+endif
+
+if cc.has_type('long long')
+  add_project_arguments('-DHAVE_LONGLONG', language: 'c')
+endif
+
+install_headers('fitsio.h', 'fitsio2.h', 'longnam.h')
+
+libcfitsio_la_SOURCES = [
+  'buffers.c',
+  'cfileio.c',
+  'checksum.c',
+  'drvrfile.c',
+  'drvrmem.c',
+  'drvrnet.c',
+  'drvrsmem.c',
+  'editcol.c',
+  'edithdu.c',
+  'eval_l.c',
+  'eval_y.c',
+  'eval_f.c',
+  'fitscore.c',
+  'getcol.c',
+  'getcolb.c',
+  'getcold.c',
+  'getcole.c',
+  'getcoli.c',
+  'getcolj.c',
+  'getcolk.c',
+  'getcoll.c',
+  'getcols.c',
+  'getcolsb.c',
+  'getcoluk.c',
+  'getcolui.c',
+  'getcoluj.c',
+  'getkey.c',
+  'group.c',
+  'grparser.c',
+  'histo.c',
+  'iraffits.c',
+  'modkey.c',
+  'putcol.c',
+  'putcolb.c',
+  'putcold.c',
+  'putcole.c',
+  'putcoli.c',
+  'putcolj.c',
+  'putcolk.c',
+  'putcoluk.c',
+  'putcoll.c',
+  'putcols.c',
+  'putcolsb.c',
+  'putcolu.c',
+  'putcolui.c',
+  'putcoluj.c',
+  'putkey.c',
+  'region.c',
+  'scalnull.c',
+  'swapproc.c',
+  'wcssub.c',
+  'wcsutil.c',
+  'imcompress.c',
+  'quantize.c',
+  'ricecomp.c',
+  'pliocomp.c',
+  'fits_hcompress.c',
+  'fits_hdecompress.c',
+  'simplerng.c',
+  'zcompress.c',
+  'zuncompress.c',
+]
+
+if has_fortran
+  libcfitsio_la_SOURCES += [
+    'f77_wrap1.c',
+    'f77_wrap2.c',
+    'f77_wrap3.c',
+    'f77_wrap4.c',
+  ]
+endif
+
+libcfitsio = shared_library(
+  'cfitsio',
+  libcfitsio_la_SOURCES,
+  dependencies: [libm, zlib, curl, threads, bzip2],
+  version: meson.project_version(),
+  soversion: SOVERSION,
+  install: true,
+)
+
+cfitsio_dep = declare_dependency(
+  include_directories: include_directories('.'),
+  link_with: libcfitsio,
+)
+
+pkg = import('pkgconfig')
+pkg.generate(
+  libraries: libcfitsio,
+  version: meson.project_version(),
+  name: 'cifitsio',
+  description: 'FITS File Subroutine Library',
+  url: 'https://heasarc.gsfc.nasa.gov/fitsio/',
+)
+
+executable(
+  'fitscopy',
+  'utilities/fitscopy.c',
+  dependencies: cfitsio_dep,
+  install: get_option('utils'),
```

Comment:
This option should be used in an "if" statement to control whether the executable() is defined -- not whether the install parameter is set.